model/DataBaseAccess.php

```diff
@@ -203,6 +236,50 @@ public function removePermissions($user, $resourceId, $rights)
         return true;
     }
 
+    /**
+     * remove batch permissions
+     *
+     * @access public
+     * @param array $data
+     * @return boolean
+     */
+    public function removeMultiplePermissions(array $data)
+    {
+        $groupedRemove = [];
+        $eventsData = [];
+        foreach ($data as $permissionItem) {
+            foreach ($permissionItem['permissions'] as $userId => $privilegeIds) {
+                if (!empty($privilegeIds)) {
+                    $groupedRemove[$userId][implode($privilegeIds)]['resources'][] = $permissionItem['resource']->getUri();
+                    $groupedRemove[$userId][implode($privilegeIds)]['privileges'] = $privilegeIds;
+                    $eventsData[] = ['userId' => $userId, 'resourceId' => $permissionItem['resource']->getUri(), 'privileges' => $privilegeIds];
+                }
+            }
+        }
+        foreach ($groupedRemove as $userId => $resources) {
+            foreach ($resources as $permissions) {
+                $inQueryPrivilege = implode(',', array_fill(0, count($permissions['privileges']), ' ? '));
+                $inQueryResources = implode(',', array_fill(0, count($permissions['resources']), ' ? '));
+                $query = sprintf(
+                    'DELETE FROM %s WHERE %s IN (%s) AND %s IN (%s) AND %s = ?',
```

Comment:
Will fail for empty `$permissions`.
IMO the case should be covered on the repository level.

Reply:
This code is rather confusing but `$permissions` can't be empty

Reply:
I'm pretty sure they can't, but we are not ensuring it.
Well, OK. That'll do for now I suppose, but a good approach would be to use objects with defined state instead of arrays.